pkg/endpoint/policy.go

```diff
@@ -751,3 +751,21 @@ func (e *Endpoint) UpdateVisibilityPolicy(annoCB AnnotationsResolverCB) {
 	}
 	<-ch
 }
+
+// GetRealizedPolicyRuleLabelsForKey returns the list of policy rule labels
+// which match a given flow key (in host byte-order)
+func (e *Endpoint) GetRealizedPolicyRuleLabelsForKey(key policy.Key) (
```

Comment:
This function was added but it seems unused. The reason I'm asking this is because this can be optimized by having a LRU map that checks if an entry exists in the that lru map and keeps returning an already deepcopied value of rules. Otherwise we are increasing the memory consumption by always returning a deepcopy of these rules.

Reply:
Yes, it's going to be used by code outside the Cilium repository. Maybe I can add a comment about this. 

But the LRU cache is a good input, I'll see if I can avoid having to `DeepCopy` everything.